torch/quantization/observer.py

```diff
@@ -63,13 +76,298 @@ def calculate_qparams(self):
 
         return torch.tensor([scale, zero_point])
 
+
+class MinMaxObserver(ObserverBase):
+    r"""Default Observer Module
+    A default implementation of the observer module, only works for
+    `per_tensor_affine` quantization scheme.  The module will record the
+    running average of max and min value of the observed Tensor and
+    calculate_qparams will calculate scale and zero_point
+    """
+
+    def __init__(self, **kwargs):
+        super(MinMaxObserver, self).__init__(**kwargs)
+        self.min_val = None
+        self.max_val = None
+
+    def forward(self, x):
+        if self.min_val is None or self.max_val is None:
+            self.min_val = torch.min(x)
+            self.max_val = torch.max(x)
+        else:
+            self.min_val = torch.min(torch.min(x), self.min_val)
+            self.max_val = torch.max(torch.max(x), self.max_val)
+
+    def calculate_qparams(self, **kwargs):
+        if self.max_val is None or self.min_val is None:
+            raise Exception("must run observer before calling calculate_qparams!")
+        return self._calculate_qparams(self.min_val.item(), self.max_val.item())
+
+
+class HistogramObserver(ObserverBase):
+    r"""
+    The module records the running histogram of tensor values along with
+    min/max values. calculate_qparams will calculate scale and zero_point
+    """
+
+    def __init__(self, bins=2048, **kwargs):
+        super(HistogramObserver, self).__init__(**kwargs)
+        self.bins = bins
+        self.histogram = None
+        self.min_val = None
+        self.max_val = None
+
+    @staticmethod
+    def _get_norm(delta_begin, delta_end, density, norm_type):
```

Comment:
Lets simplify here. Since we support only L2 norm, rewrite as:
```python
def _get_norm(delta_begin, delta_end, density):
            return (delta_end * delta_end * delta_end
                - delta_begin * delta_begin * delta_begin
            ) / 3*density
 ```